### PR TITLE
Remove AlertMacOS in favor of Alert & promptMacOS

### LINKED ISSUE #354: AlertMacOS should be deprecated in favor of Alert
Upstream react-native has `Alert` which works on iOS and Android.
`Alert` has an `options` argument that can contains a couple Android specific options.
`AlertIOS` also exists with some iOS specific options, but has been deprecated upstream.
`AlertMacOS` was created in the fork to be similar to `AlertIOS` and address macOS specific concerns.
`AlertMacOS` should now be deprecated and replaced by additional macOS specific options in the `Alert` `options` argument.

The macOS specific keys which are currently implemented under the covers in the `RCTAlertManager.alertWithArgs` method are:
```
    defaultInputs?: DefaultInputsArray,
    modal?: ?boolean,
    critical?: ?boolean,
```
The `DefaultInputsArray` is an array of objects each with:
```
  /**
   * Default input
   */
  default?: string,
  /**
   * Placeholder input
   */
  placeholder?: string,
```
to set the default text and placeholder string of the text inputs.   It can contain 1 or 2 objects: 2 objects if the alert type is `login-password`

`modal` makes the Alert appear as a modal dialog instead of a sheet.

`critical` makes the Alert icon an exclamation mark instead of the app icon.

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Copy of #662 to try and land this change.

RN Core once had a separate `Alert` and `AlertIOS`, but have since deprecated and removed the latter. We had forked `AlertIOS` into `AlertMacOS`. It's been deprecated for a while, let's just remove it to get rid of unnecessary diffs.

To this effect, I found some unnecessary example code in RN Core that we can cleanup for even less diffs. Soo overall, the order of operations is..

1) Remove AlertMacOS, and replace it's uses with `Alert.PromptMacOS` (this PR)
2) Move the examples to the `AlertIOSExample` page (this PR)
3) Made an upstream change to move `AlertIOSExamples` into `AlertExamples` (https://github.com/facebook/react-native/pull/35586/)
4) Once we merge up to that commit, sync that commit and move our macOS specific examples also into `AlertExmaples`. (Some future PR).

## Changelog

[macOS] [Deprecated] - Deprecate AlertMacOS in favor of Alert

## Test Plan

Existing RNTester tests still work.

https://user-images.githubusercontent.com/6722175/206586107-95594e43-c980-49d4-8c2c-f4fb056c1f32.mov

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/662)

### CODE REVIEW

File: Libraries/Alert/AlertMacOS.js [hunk 1]
```diff
@@ -175,6 +176,11 @@ class AlertMacOS {
     modal?: ?boolean,
     critical?: ?boolean,
   ): void {
+    warnOnce(
+      'deprecated-AlertMacOS',
+      '"AlertMacOS" module has been deprecated in favor of "Alert" and will be removed in a future version of react-native-macos',
+    );
+
```
Comment (Saadnajmi) [inline Libraries/Alert/AlertMacOS.js:183]: I checked and nobody downstream at Microsoft and on Github, and afaik nobody uses this. Maybe we can just delete it?

Comment (amgleitman) [reply Libraries/Alert/AlertMacOS.js:183]: We already warn about the usage of AlertMacOS in [index.js](https://github.com/microsoft/react-native-macos/blob/1ae2e06db25ebf32513d307aa7503c4f831cab8a/index.js#L274-L283), so this extra warning is unnecessary for standard RN development.

That being said, I'm okay with deleting this altogether. The warning's been around for [over two years](24be28ebb23f4b987f06aa6e967a9bfdd3a969bb); that's more than enough time for RN developers to migrate over.

**Review by amgleitman — DISMISSED** [review]

File: Libraries/Alert/NativeAlertManager.js [hunk 2]
```diff
@@ -28,26 +28,9 @@ export type Args = {|
...
-  cancelButtonKey?: string,
-  destructiveButtonKey?: string,
-  keyboardType?: string,
-  // [TODO(macOS GH#774)
-  defaultInputs?: Array<Object>,
-  modal?: ?boolean,
-  critical?: ?boolean,
-  // ]TODO(macOS GH#774)
-|};
-
```
[the remark this answers is not in the document]
Comment (amgleitman) [reply Libraries/Alert/NativeAlertManager.js:47]: I don't remember the exact error, but I notice the difference is that I changed the type of `defaultInputs` from `DefaultInputsArray` to `Array<Object>`, so I think it was related to that.

Any info on why codegen is breaking?

File: Libraries/Alert/AlertMacOS.js (hunk 1 above)
Comment (Saadnajmi) [reply Libraries/Alert/AlertMacOS.js:183]: Let's remove it. That would also help us clean some diffs around the test pages here. I went ahead and made a (draft) upstream fix to help with this effort: https://github.com/facebook/react-native/pull/35586

Comment (Saadnajmi) [reply Libraries/Alert/AlertMacOS.js]: Both this PR and my upstream PR are ready now. I updated my PR notes to explain how they relate.

**Comment by harrieshin** [comment]
test Sheets?

File: Libraries/Alert/Alert.js [hunk 3]
```diff
@@ -154,10 +173,53 @@ class Alert {
       // [TODO(macOS GH#774)
     } else if (Platform.OS === 'macos') {
       const defaultInputs = [{default: defaultValue}];
-      AlertMacOS.prompt(title, message, callbackOrButtons, type, defaultInputs);
+      Alert.promptMacOS(title, message, callbackOrButtons, type, defaultInputs);
     }
     // ]TODO(macOS GH#774)
   }
+
+  // [TODO(macOS GH#774)
```
Comment (harrieshin) [inline Libraries/Alert/Alert.js:181]: i feel like old code had better comments on the params?

**Comment by harrieshin** [comment]
wait in your Video sheet doesn't look like sheets?

**Comment by Saadnajmi** [comment]
> wait in your Video sheet doesn't look like sheets? 

I think Ventura made them floating windows

**Comment by harrieshin** [comment]
> > wait in your Video sheet doesn't look like sheets?
> 
> I think Ventura made them floating windows

if that's really the case, we should document expected behaviors?

**Comment by Saadnajmi** [comment]
> > > wait in your Video sheet doesn't look like sheets?
> > 
> > I think Ventura made them floating windows
> 
> if that's really the case, we should document expected behaviors? 

Once we merge to 0.71, we will have typescript types for files. When there, we can update the documentation in our repo. You can see I updated the iOS documentation in Alert.d.ts in my upstream PR here: https://github.com/facebook/react-native/pull/35586

**Review by harrieshin — APPROVED** [review]

File: Libraries/Alert/Alert.js (hunk 3 above)
Comment (Saadnajmi) [reply Libraries/Alert/Alert.js:181]: You're right. I'll see if I can copy the documentation from AlertMacOS to promptMacOS

**Pull request merged**